src/nsls2api/api/v1/proposal_api.py

```diff
@@ -298,7 +303,14 @@ async def get_slack_channels_for_proposal(
             detail=f"An error occurred: {e}",
         )
 
-    return channels
+    logger.info(f"Slack conversations for proposal {channels}")
+
+    conversations = []
+    for channel in channels:
+        conversations.append(get_conversation_details(channel.channel_id))
```

Comment:
Seems like this could be a simple list comprehension:
```suggestion
    conversations = [get_conversation_details(channel.channel_id) for channel in channels]
```